### DISCUSSION TITLE
Arch Linux Packages

### DISCUSSION THREAD

**Discussion by orhun** [sayanarijit/xplr#8] [Show and tell]
Hey, nice project!

I created the following AUR packages for installing `xplr` on Arch Linux:

* [xplr](https://aur.archlinux.org/packages/xplr/) (latest version)
* [xplr-git](https://aur.archlinux.org/packages/xplr-git/) (development version)
* [xplr-bin](https://aur.archlinux.org/packages/xplr-bin/) (latest binary)

Instructions are:

---

`xplr` can be installed from available [AUR packages](https://aur.archlinux.org/packages/?O=0&SeB=b&K=xplr&outdated=&SB=n&SO=a&PP=50&do_Search=Go) using an [AUR helper](https://wiki.archlinux.org/index.php/AUR_helpers). For example,

```
paru -S xplr
```

If you prefer, you can clone the [AUR packages](https://aur.archlinux.org/packages/?O=0&SeB=b&K=xplr&outdated=&SB=n&SO=a&PP=50&do_Search=Go) and then compile them with [makepkg](https://wiki.archlinux.org/index.php/Makepkg). For example,

```
git clone https://aur.archlinux.org/xplr.git
cd xplr/
makepkg -si
```

---

You can update the Wiki about installing from AUR if you want!

**Comment by sayanarijit** [comment]
Hey that's awesome! Thanks a lot.

**Comment by sayanarijit** [comment]
Works fine with `pamac install xplr` too. (With AUR enabled)

**Comment by gabevenberg** [comment]
looks like someone beat me to it!

**Reply by orhun** [reply]
Haha, if you want to maintain `xplr`, feel free to tell me your AUR nick so that I can add you as co-maintainer.

@sayanarijit, I can add you as well, in case you might want to setup your CI to update AUR packages on new releases one day.

**Reply by sayanarijit** [reply]
@orhun Sure. That seems like a good plan. Just registered as sayanarijit.

**Reply by orhun** [reply]
@sayanarijit I added you as co-maintainer.

**Reply by sayanarijit** [reply]
So @orhun, in order to update the AUR package automatically, should I create a PR to https://github.com/orhun/PKGBUILDs or push directly to AUR?

**Reply by orhun** [reply]
Actually no, it still requires `git push` to the AUR. See [AUR submission guidelines](https://wiki.archlinux.org/index.php/AUR_submission_guidelines) for more detail.

What you can do is setup a CD action for pulling the AUR repository (`https://aur.archlinux.org/xplr`), bumping the package (`pkgver`, `sha256sums` etc.), generating `.SRCINFO`, commiting the changes and pushing it to the AUR. I think I saw some projects do something similar.

**Comment by sayanarijit** [comment]
Hey @orhun, Since I started publishing the standalone binaries, wouldn't it be better for `xplr` AUR package to just download the binary and extract it than building it with cargo?

**Reply by orhun** [reply]
Hey @sayanarijit, unfortunately no. Base packages (e.g. `xplr`) should build it from source. If you just want to install a pre-compiled binary, you can submit a `-bin` package. Such as:

* https://aur.archlinux.org/packages/xplr-bin/

Oh wait... who submittted that package!? Oh... I did.

**Comment by sayanarijit** [comment]
@orhun There's a critical update: https://github.com/sayanarijit/xplr/releases/tag/v0.5.10

**Comment by orhun** [comment]
Hey @sayanarijit, I moved `xplr` package to the [community repository](https://wiki.archlinux.org/title/Official_repositories#community):

https://archlinux.org/packages/community/x86_64/xplr/

which means that now it can be installed directly with pacman:

 ```
pacman -S xplr
```

AUR package (https://aur.archlinux.org/packages/xplr/) will be deleted soon. You might want to update your installation instructions :)

**Reply by sayanarijit** [reply]
Great! Thanks..

**Reply by sayanarijit** [reply]
Updated the wiki.
